File: src/php/inserttask.php
<?php 
	require('init_function.php');
	session_start();
	$taskid = getNextTaskId();
	$taskname = $_POST['textTaskName'];
	$useractive = $_SESSION['userlistapp'];
	$createddate = date('Y-m-d H:i:s');
	$deadlinedate = $_POST['textDeadline']." ".$_POST['textTimeDeadline'].":00";
	echo $deadlinedate;
	$status = "UNCOMPLETE";
	$categoryid = $_GET['categoryid'];
	$listAssignee = $_POST['textAssignee'];
	$listTag = $_POST['textTag'];
	
	echo "Task ID : ".$taskid."<br />";
	echo "User Active : ".$useractive."<br />";
	echo "Create Date : ".$createddate."<br />";
	echo "Deadline Date : ".$deadlinedate."<br />";
	echo "Task Name : ".$taskname."<br />";
	echo "Status : ".$status."<br />";
	echo "Task ID : ".$categoryid."<br />";
	echo "List Assigne : ".$listAssignee."<br />";
	echo "List Tag : ".$listTag."<br />";
	
	$con = getConnection();
	$query = "INSERT INTO `task` 
	(`taskid`, `taskname`, `username`, `createddate`, `deadline`, `status`, `categoryid`) VALUES 
	($taskid, '$taskname', '$useractive', '$createddate', '$deadlinedate', '$status', $categoryid)";
	mysqli_query($con,$query);
	
	$query = "INSERT INTO `assignee` (`username`, `taskid`) VALUES ('$useractive', $taskid)";
	mysqli_query($con,$query);
	$Assignee = explode(',', $listAssignee);
	for($i = 0; $i < count($Assignee) ; $i++){
			$query = "INSERT INTO `assignee` (`username`, `taskid`) VALUES ('$Assignee[$i]', $taskid)";
			mysqli_query($con,$query);
	}
	
	$Tag = explode(',', $listTag);
	for($i = 0; $i < count($Tag) ; $i++){
			$tagid = getTagId($Tag[$i]);
			$query = "INSERT INTO `task_tag` (`taskid`, `tagid`) VALUES ($taskid, ".$tagid.");";
			mysqli_query($con,$query);
	}
	
	if(isset($_FILES["attachment1"]["size"])){
		$file = $_FILES['attachment1'];
		upload_attachment($file,$useractive);
		$filename = $file['name'];
		if(strcmp($filename,"") != 0){
			$query = "INSERT INTO `attachment` (`filename`, `taskid`) VALUES ('$useractive-$filename', $taskid)";
			mysqli_query($con,$query);
		}
	}
	if(isset($_FILES["attachment2"]["size"])){
		$file = $_FILES['attachment2'];
		upload_attachment($file,$useractive);
		$filename = $file['name'];
		if(strcmp($filename,"") != 0){
			$query = "INSERT INTO `attachment` (`filename`, `taskid`) VALUES ('$useractive-$filename', $taskid)";
			mysqli_query($con,$query);
		}
	}
	if(isset($_FILES["attachment3"]["size"])){
		$file = $_FILES['attachment3'];
		upload_attachment($file,$useractive);
		$filename = $file['name'];
		if(strcmp($filename,"") != 0){
			$query = "INSERT INTO `attachment` (`filename`, `taskid`) VALUES ('$useractive-$filename', $taskid)";
			mysqli_query($con,$query);
		}
	}
	if(isset($_FILES["attachment4"]["size"])){
		$file = $_FILES['attachment4'];
		upload_attachment($file,$useractive);
		$filename = $file['name'];
		if(strcmp($filename,"") != 0){
			$query = "INSERT INTO `attachment` (`filename`, `taskid`) VALUES ('$useractive-$filename', $taskid)";
			mysqli_query($con,$query);
		}
	}
	if(isset($_FILES["attachment5"]["size"])){
		$file = $_FILES['attachment5'];
		upload_attachment($file,$useractive);
		$filename = $file['name'];
		if(strcmp($filename,"") != 0){
			$query = "INSERT INTO `attachment` (`filename`, `taskid`) VALUES ('$useractive-$filename', $taskid)";
			mysqli_query($con,$query);
		}
	}
	//header("Location: ../page/dashboard.php");
?>

File: TEBTaskManage/src/Servlet/inserttask.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Servlet;

import Class.Function;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.PrintWriter;
import java.sql.Connection;
import java.sql.Statement;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Iterator;
import java.util.List;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 *
 * @author User
 */
public class inserttask extends HttpServlet {

    /**
     * Processes requests for both HTTP
     * <code>GET</code> and
     * <code>POST</code> methods.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        response.setContentType("text/html;charset=UTF-8");
        PrintWriter out = response.getWriter();
        try {
//            Boolean isMultipartContent = ServletFileUpload.isMultipartContent(request);
//            if(!isMultipartContent){
//                out.print("you not gona upload file!!");
//            }else{
//                out.print("you try to upload file!!");
//            }
//            
//            String taskName = "";
//            String deadline = "";
//            String timeDeadline = "";
//            String listAssignee = "";
//            String listTag = "";
//            String taskId = (new Function()).GetNextTaskId();
//            String userActive = "";
//            String categoryId = request.getParameter("categoryid");
//            
//            if(request.getSession().getAttribute("userlistapp")!=null){
//                userActive = request.getSession().getAttribute("userlistapp").toString();
//            }
//            
//            DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
//            Date date = new Date();
//            String createdDateTime = dateFormat.format(date);
//            
//            
//            GetConnection getCon = new GetConnection();
//            Connection conn = getCon.getConnection();
//            String query = "INSERT INTO `task` " +
//            "(`taskid`, `taskname`, `username`, `createddate`, `deadline`, `status`, `categoryid`) VALUES "+
//            "("+taskId+", 'aaa', '"+userActive+"', '1111-11-11 11:11:11', '1111-11-11 11:11:11', 'UNCOMPLETE', "+categoryId+")";
//            Statement stt = conn.createStatement();
//            stt.execute(query);
//            
//            FileItemFactory factory = new DiskFileItemFactory();
//            ServletFileUpload upload = new ServletFileUpload(factory);
//            List<FileItem> fields = upload.parseRequest(request);
//            out.println("Number of fields: " + fields.size());
//            Iterator<FileItem> it = fields.iterator();
//            if (!it.hasNext()) {
//                    out.println("No fields found");
//                    return;
//            }
//            while (it.hasNext()) {
//                    FileItem fileItem = it.next();
//                    boolean isFormField = fileItem.isFormField();
//                    if (isFormField) {
//                            out.print("<br />"+fileItem.getFieldName()+" : "+fileItem.getString());
//                            if(fileItem.getFieldName().toString().equals("textTaskName"))
//                                taskName = fileItem.getString();
//                            else if(fileItem.getFieldName().toString().equals("textDeadline"))
//                                deadline = fileItem.getString();
//                            else if(fileItem.getFieldName().toString().equals("textTimeDeadline"))
//                                timeDeadline = fileItem.getString();
//                            else if(fileItem.getFieldName().toString().equals("textAssignee"))
//                                listAssignee = fileItem.getString();
//                            else if(fileItem.getFieldName().toString().equals("textTag"))
//                                listTag = fileItem.getString();
//                    } else {
//                            String fileName = fileItem.getName().toString();
//                            if(!fileName.equals("")){
//                                out.print("<br />"+fileName);
//                                int index = fileName.lastIndexOf('.') + 1;
//                                String extension = fileName.substring(index);
//
//                                byte [] arByte = fileItem.get();
//                                String dir = request.getRealPath("attachment");
//                                FileOutputStream fileOutStream = new FileOutputStream(dir+"/"+taskId+"-"+fileName);
//                                fileOutStream.write(arByte);
//                                fileOutStream.close();
//                                
//                                getCon = new GetConnection();
//                                conn = getCon.getConnection();
//                                query = "INSERT INTO `attachment` " +
//                                " VALUES "+
//                                    "('"+taskId+"-"+fileName+"',"+taskId+")";
//                                stt = conn.createStatement();
//                                stt.execute(query);
//                            }
//                    }
//            }
//            
//            deadline = deadline+" "+timeDeadline+":00";
//            getCon = new GetConnection();
//            conn = getCon.getConnection();
//            query = "UPDATE `task` SET" +
//                    " taskname='"+taskName+"'"+
//                    ",createddate='"+createdDateTime+"'"+
//                    ",deadline='"+deadline+"'"+
//                    " WHERE taskid="+taskId;
//            System.out.println(query);
//            stt = conn.createStatement();
//            stt.execute(query);
//            
//            query = "INSERT INTO `assignee` (`username`, `taskid`) VALUES ('"+userActive+"', "+taskId+")";
//            stt.execute(query);
//            String [] assignee = listAssignee.split(",");
//            for(int i = 0; i < assignee.length ; i++){
//                query = "INSERT INTO `assignee` (`username`, `taskid`) VALUES ('"+assignee[i]+"', "+taskId+")";
//                stt.execute(query);
//            }
//            
//            String [] tag = listTag.split(",");
//            for(int i = 0; i < tag.length ; i++){
//                String tagid = (new Function()).GetTagId(tag[i]);
//                query = "INSERT INTO `task_tag` (`taskid`, `tagid`) VALUES ("+taskId+", "+tagid+");";
//                stt.execute(query);
//            }
//            
//            response.sendRedirect("dashboard.jsp");
//            conn.close();
        } catch(Exception exc){
            System.out.println("Error : "+exc.toString());
        }finally {            
            out.close();
        }
    }

    // <editor-fold defaultstate="collapsed" desc="HttpServlet methods. Click on the + sign on the left to edit the code.">
    /**
     * Handles the HTTP
     * <code>GET</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        processRequest(request, response);
    }

    /**
     * Handles the HTTP
     * <code>POST</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        processRequest(request, response);
    }

    /**
     * Returns a short description of the servlet.
     *
     * @return a String containing servlet description
     */
    @Override
    public String getServletInfo() {
        return "Short description";
    }// </editor-fold>
}
